File: 2421-maximum-number-of-pairs-in-array/2421-maximum-number-of-pairs-in-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var numberOfPairs = function(nums) {
    let pairs = 0;
    const map = {};

    for (let i = 0; i < nums.length; i++) {
        if (map[nums[i]]) {
            delete map[nums[i]];
            pairs++;
        } else {
            map[nums[i]] = 1;
        }
    }

    return [pairs, Object.values(map).length];
};

File: 2421-maximum-number-of-pairs-in-array/2421-maximum-number-of-pairs-in-array.py
class Solution:
    def numberOfPairs(self, nums: List[int]) -> List[int]:
        pairs, hashMap = 0, {}

        for num in nums:
            if num in hashMap:
                del hashMap[num]
                pairs += 1
            else:
                hashMap[num] = 1
        
        return [pairs, len(hashMap)]